frame/bridge/src/lib.rs

```diff
@@ -136,67 +139,84 @@ impl<T: Trait> Module<T> {
 			*state_root,
 			proof.clone()
-		)?;
+		);
+
+		// TODO: Remove once I can impl From<StorageError> for DispatchError
+		let checker = match checker {
+			Ok(c) => c,
+			Err(e) => return Self::map_storage_err(e).into()
+		};
 
 		// By encoding the given set we should have an easy way to compare
 		// with the stuff we get out of storage via `read_value`
 		let encoded_validator_set = validator_set.encode();
-		let actual_validator_set = checker
-			.read_value(b":grandpa_authorities")?
-			.ok_or(Error::StorageValueUnavailable)?;
+
+		// TODO: Remove once I can impl From<StorageError> for DispatchError
+		let actual_validator_set = match checker.read_value(b":grandpa_authorities") {
+			Ok(c) => c.ok_or(Error::<T>::StorageValueUnavailable)?,
+			Err(e) => return Self::map_storage_err(e).into()
+		};
 
 		if encoded_validator_set == actual_validator_set {
 			Ok(())
 		} else {
-			Err(Error::ValidatorSetMismatch)
+			Err(Error::<T>::ValidatorSetMismatch.into())
 		}
 	}
-}
 
-// A naive way to check whether a `child` header is a decendent
-// of an `ancestor` header. For this it requires a proof which
-// is a chain of headers between (but not including) the `child`
-// and `ancestor`. This could be updated to use something like
-// Log2 Ancestors (#2053) in the future.
-fn verify_ancestry<H>(proof: Vec<H>, ancestor_hash: H::Hash, child: H) -> Result<(), Error>
-where
-	H: Header
-{
-	let mut parent_hash = child.parent_hash();
-
-	// If we find that the header's parent hash matches our ancestor's hash we're done
-	for header in proof.iter() {
-		// Need to check that blocks are actually related
-		if header.hash() != *parent_hash {
-			break;
+	// A naive way to check whether a `child` header is a decendent
+	// of an `ancestor` header. For this it requires a proof which
+	// is a chain of headers between (but not including) the `child`
+	// and `ancestor`. This could be updated to use something like
+	// Log2 Ancestors (#2053) in the future.
+	fn verify_ancestry<H>(proof: Vec<H>, ancestor_hash: H::Hash, child: H) -> DispatchResult
+	where
+		H: Header,
+	{
+		let mut parent_hash = child.parent_hash();
+
+		// If we find that the header's parent hash matches our ancestor's hash we're done
+		for header in proof.iter() {
+			// Need to check that blocks are actually related
+			if header.hash() != *parent_hash {
+				break;
+			}
+
+			parent_hash = header.parent_hash();
+			if *parent_hash == ancestor_hash {
+				return Ok(())
+			}
 		}
 
-		parent_hash = header.parent_hash();
-		if *parent_hash == ancestor_hash {
-			return Ok(())
-		}
+		Err(Error::<T>::InvalidAncestryProof.into())
 	}
 
-	Err(Error::InvalidAncestryProof)
+	// TODO: Remove once I can impl From<StorageError> for DispatchError
+	fn map_storage_err(e: StorageError) -> DispatchError {
+		match e {
+			StorageError::StorageRootMismatch => Error::<T>::StorageRootMismatch.into(),
+			StorageError::StorageValueUnavailable => Error::<T>::StorageValueUnavailable.into(),
+		}
```

Comment:
```suggestion
		}.into()
```
You could put `into` here to avoid duplication, but it's a minor thing.